libraries/lib-time-and-pitch/StaffPad/TimeAndPitch.cpp

```diff
@@ -43,16 +44,17 @@ inline float lagrange6(const float (&smp)[6], float t)
   return (c0 + c1 * t) + (c2 + c3 * t) * t2 + (c4 + c5 * t) * t2 * t2;
 }
 
-inline int getFftSize(int sampleRate)
+std::vector<std::complex<float>> generateRandomPhaseVector(size_t size)
 {
-  // 44.1kHz maps to 4096 samples (i.e., 93ms).
-  // We grow the FFT size proportionally with the sample rate to keep the
-  // window duration roughly constant, with quantization due to the
-  // power-of-two constraint.
-  // If needed some time in the future, we can decouple analysis window and
-  // FFT sizes by zero-padding, allowing for very fine-grained window duration
-  // without compromising performance.
-  return 1 << (12 + (int)std::round(std::log2(sampleRate / 44100.)));
+  // keep the processing deterministic
```

Comment:
I don't understand this point. Why do we need it deterministic? How much do we care about this determinism? 

Provided that `srand` modifies the global state I wonder if something like https://en.cppreference.com/w/cpp/numeric/random/linear_congruential_engine should be used instead.

Reply:
Sounds good, I'll try.

Reply:
> I don't understand this point. Why do we need it deterministic? How much do we care about this determinism?

For A/B comparison before and after some algorithmic tweaks, for example. I might want to make sure that my modifications do affect the signal, so I render it once before and once after. If the process weren't deterministic I couldn't tell.